GettingStartedWithInterfacesAndMocksTests/DieExtensionsTests.cs

```diff
@@ -25,5 +25,17 @@ public void IsValidRoll_RollHas5Dice_ReturnsTrue()
 
             dice.IsValidRoll().Should().BeTrue(because: "A valid game roll has five dice");
         }
+
+        [TestMethod]
+        public void IsFiveOfAKind_AllDiceAreTheSame_ReturnsTrue()
+        {
+            IEnumerable<Die> dice = null; //Uh oh... how do I create dice that are all the same?
+        }
```

Comment:
Uh-oh, I've run into a problem. The DIe class randomly generates its value when it is created or when I roll it. That value is read only (as I will maintain that it should be), so how can I create an IEnumerable<Die> where the dice all have the same value? This is one of the reasons that objects that use randomness can be difficult to test. This is where interfaces can come to our rescue.

Let's do a little refactoring...